### PR TITLE
:book: Add docs for cluster stack operator

### LINKED ISSUE #27: Documentation about how to use custom resources
/kind feature

**Describe the solution you'd like**
There exists the https://github.com/SovereignCloudStack/cluster-stacks-demo that makes it easy with two make targets to create a workload cluster with docker.

However, there is no documentation yet on how the actual process of dealing with the cluster-stack-operator looks like. 

1. Create ClusterStack and maybe the ProviderClusterStackReleaseTemplate
2. Wait until a release of a cluster stack is ready
3. Create a cluster with the ClusterClass feature of Cluster API by referencing the ClusterClass corresponding to the respective version of a cluster stack.

These three steps should be documented appropriately. Additionally, there should be some hints on where to look in order to get more information about the system.

### PR DESCRIPTION
**Pull request by janiskemper**
**What this PR does / why we need it**:
Adding docs to understand, develop, and use the cluster stacks and the cluster stack operator.

**Which issue(s) this PR fixes**:
Fixes #27

**TODOs**:
- [ ] squash commits
- [x] include documentation
- [ ] add unit tests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,86 +8,27 @@ The operator can be used with any repository that contains releases of cluster s
...
-3. Node images: Node images can be provided to the user in different form. They are released and tested together with the other two components of the cluster stack.
+The Cluster Stacks give an answer by working hand-in-hand with Cluster API to facilitate self-service Kubernetes. They provide a framework and tools for managing a fully open-source self-service Kubernetes infrastructure with ease. They integrate seamlessly in the Cluster API cosmos. 
 
-The first two are handled by this operator here. The node images, on the other hand, have to be handled by separate provider integrations, similar to the ones that [Cluster-API uses](https://cluster-api.sigs.k8s.io/developer/providers/implementers-guide/overview).
+The Cluster Stack operator enables an “Infrastructure as Software” approach for managing Kubernetes clusters in self-service.
 
-## Implementing a provider integration
+The cluster stacks are very generic and can be adapted to many use cases.
```
Comment (jschoone) [inline README.md:21]:
```suggestion
The Cluster Stacks are very generic and can be adapted to many use cases. 
```

File: README.md [hunk 2]
```diff
@@ -8,86 +8,27 @@ The operator can be used with any repository that contains releases of cluster s
...
+The Cluster Stacks give an answer by working hand-in-hand with Cluster API to facilitate self-service Kubernetes. They provide a framework and tools for managing a fully open-source self-service Kubernetes infrastructure with ease. They integrate seamlessly in the Cluster API cosmos. 
 
-The first two are handled by this operator here. The node images, on the other hand, have to be handled by separate provider integrations, similar to the ones that [Cluster-API uses](https://cluster-api.sigs.k8s.io/developer/providers/implementers-guide/overview).
+The Cluster Stack operator enables an “Infrastructure as Software” approach for managing Kubernetes clusters in self-service.
 
-## Implementing a provider integration
+The cluster stacks are very generic and can be adapted to many use cases. 
 
-Further information and documentation on how to implement a provider integration will follow soon.
+### Are cluster stacks relevant to you?
```
Comment (jschoone) [inline README.md:23]:
```suggestion
### Are Cluster Stacks relevant to you?
```

File: README.md [hunk 3]
```diff
@@ -8,86 +8,27 @@ The operator can be used with any repository that contains releases of cluster s
...
 
-## Developing Cluster Stack Operator
+Are you interested in setting up Kubernetes in your company based on open-source software? Do you not want to rely on other providers but own your Kubernetes clusters? Do you want to manage Kubernetes clusters for others? Do you plan on using Cluster-API?
 
-Developing our operator is quite easy. First, you need to install some base requirements: Docker and Go. Second, you need to configure your environment variables. Then you can start developing with the local Kind cluster and the Tilt UI to create a workload cluster that is already pre-configured. 
+In all of these cases, the cluster stacks are for you!
```
Comment (jschoone) [inline README.md:27]:
```suggestion
In all of these cases, the Cluster Stacks are for you! 
```

File: README.md [hunk 4]
```diff
@@ -8,86 +8,27 @@ The operator can be used with any repository that contains releases of cluster s
...
+The Cluster Stack operator enables an “Infrastructure as Software” approach for managing Kubernetes clusters in self-service.
 
-## Implementing a provider integration
+The cluster stacks are very generic and can be adapted to many use cases. 
 
-Further information and documentation on how to implement a provider integration will follow soon.
+### Are cluster stacks relevant to you?
 
-## Developing Cluster Stack Operator
+Are you interested in setting up Kubernetes in your company based on open-source software? Do you not want to rely on other providers but own your Kubernetes clusters? Do you want to manage Kubernetes clusters for others? Do you plan on using Cluster-API?
```
Comment (jschoone) [inline README.md:25]:
```suggestion
Are you interested in setting up Kubernetes in your company based on open-source software? Do you not want to rely on other providers but own your Kubernetes clusters? Do you want to manage Kubernetes clusters for others? Do you plan on using Cluster API?
```

**Review by jschoone — APPROVED** [review]
Just little things for a uniform shape.

File: docs/architecture/overview.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+The node images, on the other hand, have to be handled by separate provider integrations, similar to the ones that [Cluster-API uses](https://cluster-api.sigs.k8s.io/developer/providers/implementers-guide/overview).
+
+## Cluster Stack Provider Integrations
+
+The Cluster Stack Operator is accompanied by Cluster Stack Provider Integrations. A provider integration is also an operator that works together with the Cluster Stack Operator in a specific way, which is described in the docs about building [provider integrations](../develop/provider-integration.md). 
+
+A provider integration makes sure that the node images are taken care of and made available to the user.
+
+If there is no work to be done for node images, then the Cluster Stack Operator can work in `noProvider` mode and this Cluster Stack Provider Integration can be omitted.
```
Comment (mxmxchere) [inline docs/architecture/overview.md:43]: thats handy👌🏻, good to know

**Comment by mxmxchere** [comment]
This was a helpful document that explained the scope of the operator very good while being not too long :)

This feedback might be late to the game but i think rephrasing of provider-specific code would make it even easier to understand the whole concept:

Provider could be renamed to something like "ImageController" or "ImageManager" or "ImageUploader". Its whole purpose is to upload and manage images in a specific cloud.

In [the clusterstack spec](https://github.com/SovereignCloudStack/cluster-stack-operator/blob/docs-folder/docs/reference/clusterstack.md#overview-of-clusterstackspecl) we have three fields that have "provider" in it (`provider`, `noProvider` and `providerRef`). If all three of them were removed and we had only one like "ImageController" we would get:

ImageController: none
or
ImageController: openstack

instead of three fields that seem redundant to me.

I really appreciate that it is optional to use the imagecontroller. I think this is a good design not to have a dependency between clusteraddons and the imagecontroller.

**Review by mxmxchere — APPROVED** [review]

**Pull request merged**